pkg/proxy/conntrack/conntrack.go

```diff
@@ -49,95 +50,143 @@ type Interface interface {
 	}
 	return nil
 }
 
-// ClearEntriesForNAT is part of Interface
-func (ct *execCT) ClearEntriesForNAT(origin, dest string, protocol v1.Protocol) error {
-	parameters := parametersWithFamily(utilnet.IsIPv6String(origin), "-D", "--orig-dst", origin, "--dst-nat", dest,
-		"-p", protoStr(protocol))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
+// ClearEntriesForNAT delete the conntrack entries for connections specified by the
+// destination IP(original direction) and source IP(reply direction).
+func (ct *conntracker) ClearEntriesForNAT(origin, dest string, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstIP: netutils.ParseIPSloppy(origin),
+		},
+		reply: &connectionTuple{
+			srcIP: netutils.ParseIPSloppy(dest),
+		},
+	}
+
+	klog.V(4).InfoS("Clearing conntrack entries", "origin", origin, "destination", dest, "protocol", protocol)
+	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, getNetlinkFamily(netutils.IsIPv6String(origin)), filter)
+	if err != nil {
 		// TODO: Better handling for deletion failure. When failure occur, stale udp connection may not get flushed.
 		// These stale udp connection will keep black hole traffic. Making this a best effort operation for now, since it
 		// is expensive to baby sit all udp connections to kubernetes services.
-		return fmt.Errorf("error deleting conntrack entries for UDP peer {%s, %s}, error: %v", origin, dest, err)
+		return fmt.Errorf("error deleting conntrack entries for %s peer {%s, %s}, error: %w", protoStr(protocol), origin, dest, err)
 	}
+	klog.V(4).InfoS("Cleared conntrack entries", "count", n)
 	return nil
 }
 
-// ClearEntriesForPortNAT is part of Interface
-func (ct *execCT) ClearEntriesForPortNAT(dest string, port int, protocol v1.Protocol) error {
+// ClearEntriesForPortNAT delete the conntrack entries for connections specified by the
+// destination Port(original direction) and source IP(reply direction).
+func (ct *conntracker) ClearEntriesForPortNAT(dest string, port int, protocol v1.Protocol) error {
 	if port <= 0 {
 		return fmt.Errorf("wrong port number. The port number must be greater than zero")
 	}
-	parameters := parametersWithFamily(utilnet.IsIPv6String(dest), "-D", "-p", protoStr(protocol), "--dport", strconv.Itoa(port), "--dst-nat", dest)
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
-		return fmt.Errorf("error deleting conntrack entries for UDP port: %d, error: %v", port, err)
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstPort: uint16(port),
+		},
+		reply: &connectionTuple{
+			srcIP: netutils.ParseIPSloppy(dest),
+		},
+	}
+	klog.V(4).InfoS("Clearing conntrack entries", "destination", dest, "port", port, "protocol", protocol)
```

Comment:
same comment as above, destination makes me thing about the original destination, this is a about the natted packet